### ISSUE TITLE
RenderListData response processing issue

### ISSUE THREAD

**Issue by koltyakov** [SharePoint/PnP-JS-Core#671]
#### Category
[ ] Enhancement

[x] Bug

[ ] Question

#### Version

Please specify what version of the library you are using: [3.0.3]

#### Observed Behavior

List's `renderListData` method fails while parsing a response.

```typescript
let items = await pnp.sp.web.lists.getByTitle('MyList')
    .renderListData(`
      <View>
        <Query>
          <ViewFields>
            <FieldRef Name="Title" />
          </ViewFields>
        </Query>
      </View>
    `);
```

![image](https://user-images.githubusercontent.com/7816483/34247935-59f20922-e645-11e7-8a6e-5907e766a37d.png)

Response data is an object, not a string. Checked this in SPO and On-Prem 2013, also with external lists responses.

**Comment by patrick-rodgers** [comment]
Is this in verbose json mode? Some of the methods have different results like this and we try and parse those correctly to normalize the responses.

**Comment by koltyakov** [comment]
Yep, it was in verbose mode cause I tested fairly identical scripts with 2013 (without OData extensions) and SPO.
Let it stay for a while in investigating mode, will test with no metadata and minimal metadata too over this weekend.

**Comment by patrick-rodgers** [comment]
ok - this is fairly common, you can see how we handled [it here for example](https://github.com/SharePoint/PnP-JS-Core/blob/master/src/sharepoint/regionalsettings.ts#L91)

**Comment by patrick-rodgers** [comment]
Fixed with the merge of #672. Closing this as it is fixed, please let us know if that is not the case.
